src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -756,7 +756,8 @@ AdaptiveExecutor(CitusScanState *scanState)
 		DoRepartitionCleanup(jobIdList);
 	}
 
-	if (SortReturning && distributedPlan->hasReturning)
+	if (SortReturning && distributedPlan->expectResults && job->jobQuery->commandType !=
```

Comment:
Like above, only here it's especially bad if we sort what might already have applied an `ORDER BY`